nebu/tests/models/test_utils.py

```diff
@@ -68,3 +75,89 @@ def test_convert_to_model_compat_metadata():
         'version': '1.9',
     }
     assert converted_metadata == expected_metadata
+
+
+def test_id_from_metadata():
+    assert id_from_metadata({}) is None
+    id = 'foo@bar'
+    assert id_from_metadata({'cnx-archive-uri': id}) == id
+
+
+class TestScanForIdMapping(object):
+
+    def relpaths(self, i, start):
+        k, v = i
+        v = v.relative_to(start)
+        return (k, v,)
+
+    def make_results_relative(self, results, start):
+        return dict(
+            map(
+                partial(self.relpaths, start=start),
```

Comment:
This is a pretty complicated way of making the results relative :open_mouth: 

Isn't it easier to read if you do this instead?

```python
results = {k: v.relative_to(start)
           for k, v in results.items()}
```